### PR TITLE
Add new aws-vault formula

### PR DESCRIPTION
**Pull request by aussiegeek**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/aws-vault.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  desc "securely stores and accesses AWS credentials in dev environments"
+  homepage "https://github.com/99designs/aws-vault"
+  url "https://github.com/99designs/aws-vault/archive/v3.3.0.tar.gz"
+  sha256 "2eb7ad97de31d55d61c87b26c7f361aa49f3386ee63796930373a57ef53053ed"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GO15VENDOREXPERIMENT"] = "1"
```
Comment (DomT4) [inline Formula/aws-vault.rb]: Don't need this line. It's on by default since Go 1.6.

File: Formula/aws-vault.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    srcpath = buildpath/"src/github.com/99designs/aws-vault"
+    srcpath.install Dir["*"]
+
+    cd srcpath do
+      system "go", "build"
+      bin.install "aws-vault"
```
Comment (DomT4) [inline Formula/aws-vault.rb]: Might want a `prefix.install_metafiles` line underneath this one.

**Comment by DomT4** [comment]
Ah, I thought this sounded familiar. The upstream developer has asked we don't build this from source [before](https://github.com/Homebrew/homebrew-binary/pull/317) but rather use the upstream executables.

I suggested adding it to `Caskroom/homebrew-cask` which seemed agreeable to all, so it may well be worth pursuing that method instead.

**Comment by aussiegeek** [comment]
Seems like a good argument for a binary build, closing

**Pull request closed without merging**